backend/favorite.go

```diff
@@ -41,31 +43,62 @@ func handleFavorite(w http.ResponseWriter, r *http.Request) {
 }
 
 func getFavorite(w http.ResponseWriter, r *http.Request) {
-	favorite := readFavoriteFromCookie(r)
+	favorite := readFavoriteFromCookie(r, AMP_FAVORITE_COOKIE)
 	writeFavorite(w, favorite)
 }
 
 func setFavorite(w http.ResponseWriter, r *http.Request) {
+	favorite := !readFavoriteFromCookie(r, AMP_FAVORITE_COOKIE)
+	writeFavoriteCookie(w, r, AMP_FAVORITE_COOKIE, favorite)
+	writeFavorite(w, favorite)
+}
+
+func handleFavoriteWithCount(w http.ResponseWriter, r *http.Request) {
+	EnableCors(w, r)
+	SetContentTypeJson(w)
+	SetMaxAge(w, 0)
+	if r.Method == "POST" {
+		setFavoriteWithCount(w, r)
+	} else if r.Method == "GET" {
+		getFavoriteWithCount(w, r)
+	}
+}
+
+func getFavoriteWithCount(w http.ResponseWriter, r *http.Request) {
+	favorite := readFavoriteFromCookie(r, AMP_FAVORITE_COUNT_COOKIE)
+	writeFavoriteWithCount(w, favorite)
+}
+
+func setFavoriteWithCount(w http.ResponseWriter, r *http.Request) {
+	favorite := !readFavoriteFromCookie(r, AMP_FAVORITE_COUNT_COOKIE)
+	writeFavoriteCookie(w, r, AMP_FAVORITE_COUNT_COOKIE, favorite)
+	writeFavoriteWithCount(w, favorite)
+}
+
+func writeFavoriteWithCount(w http.ResponseWriter, favorite bool) {
+	response := fmt.Sprintf("{ \"value\": %t, \"count\": 123}", favorite)
+	w.Write([]byte(response))
+}
+
+func writeFavorite(w http.ResponseWriter, favorite bool) {
+	response := fmt.Sprintf("%t", favorite)
+	w.Write([]byte(response))
+}
+
+func writeFavoriteCookie(w http.ResponseWriter, r *http.Request, name string, value bool) {
```

Comment:
on top of my head: this file is called favorite.go and yet every method name has "favorite". For example this `writeFavoriteCookie`, couldn't just be writeCookie?

Reply:
These methods are visible across all go files -> writeCookie is too generic